Sources/TuistGenerator/Generator/SchemesGenerator.swift

```diff
@@ -149,87 +204,83 @@ final class SchemesGenerator: SchemesGenerating {
-    ///
-    /// - Parameters:
-    ///   - testAction: test actions.
-    ///   - project: Project manifest.
-    ///   - generatedProject: Generated Xcode project.
-    /// - Returns: Array of buildable references.
-    private func testCoverageTargetReferences(testAction: TestAction, project: Project, generatedProject: GeneratedProject) -> [XCScheme.BuildableReference] {
-        var codeCoverageTargets: [XCScheme.BuildableReference] = []
-        testAction.codeCoverageTargets.forEach { name in
-
-            guard let target = project.targets.first(where: { $0.name == name }) else { return }
-            guard let pbxTarget = generatedProject.targets[name] else { return }
-
-            let reference = self.targetBuildableReference(target: target,
-                                                          pbxTarget: pbxTarget,
-                                                          projectName: generatedProject.name)
-            codeCoverageTargets.append(reference)
-        }
-
-        return codeCoverageTargets
+                                      preActions: preActions,
+                                      postActions: postActions)
     }
 
     /// Generates the scheme test action.
     ///
     /// - Parameters:
     ///   - scheme: Scheme manifest.
-    ///   - project: Project manifest.
-    ///   - generatedProject: Generated Xcode project.
+    ///   - graph: Tuist graph.
+    ///   - rootPath: Root path to either project or workspace.
+    ///   - generatedProjects: Project paths mapped to generated projects.
     /// - Returns: Scheme test action.
     func schemeTestAction(scheme: Scheme,
-                          project: Project,
-                          generatedProject: GeneratedProject) -> XCScheme.TestAction? {
+                          graph: Graphing,
+                          rootPath: AbsolutePath,
+                          generatedProjects: [AbsolutePath: GeneratedProject]) throws -> XCScheme.TestAction? {
         guard let testAction = scheme.testAction else { return nil }
-
+        
         var testables: [XCScheme.TestableReference] = []
         var preActions: [XCScheme.ExecutionAction] = []
         var postActions: [XCScheme.ExecutionAction] = []
 
-        testAction.targets.forEach { name in
-            guard let target = project.targets.first(where: { $0.name == name }), target.product.testsBundle else { return }
-            guard let pbxTarget = generatedProject.targets[name] else { return }
-
-            let reference = self.targetBuildableReference(target: target,
-                                                          pbxTarget: pbxTarget,
-                                                          projectName: generatedProject.name)
+        try testAction.targets.forEach { testActionTarget in
+            let pathToProject = testActionTarget.projectPath
+            guard let target = try graph.target(path: pathToProject, name: testActionTarget.name) else { return }
```

Comment:
suggestion:

seems we have a recurring operation where given `TargetReferernce` we obtain a `buildableReference` - perhaps we can create another helper method? That will reduce receptions and ensure if we fix something (e.g. like the inconsistency I mentioned earlier) we'd have a single place to do so.